### PR TITLE
Add generic trace property in requirements

### LINKED ISSUE #948: Add generic trace to requirements after all
Requirements elements should have an option to connect system models without providing semantic

### PR DESCRIPTION
**Pull request by franzTobiasDLR**
Enable generic tracing of requirements to system model elements.

All requirements can now have a generic link to system model artifcats.

Closes #948

### CODE REVIEW

File: de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/provider/RequirementsAttributeLabelProvider.java [hunk 1]
```diff
@@ -94,23 +96,33 @@ public String getColumnText(Object object, int columnIndex) {
...
 				ATypeInstance ti = valueSwitch.doSwitch(propertyInstance);
 				redirectNotification(ti, object);
 				
-				return getVerificationLabel(new Requirement(ca));
+				return getTracingLabel(new Requirement(ca));
 				
 
-			} else if (columnIndex > STATUS_COLUMN) {
+			} else if (columnIndex >= UiSnippetCustomRequirementsAttributeTable.FIXED_COLUMNS_NUMBER) {
 				APropertyInstance propertyInstance = ca.getPropertyInstances()
```
Comment (maurpa) [inline de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/provider/RequirementsAttributeLabelProvider.java:120]: APropertyInstance propertyInstance = ca.getPropertyInstances().get(REQUIREMENT_TRACE_PROPERTY_NUMBER); Duplicate in else if

File: de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/provider/RequirementsAttributeLabelProvider.java [hunk 2]
```diff
@@ -94,23 +96,33 @@ public String getColumnText(Object object, int columnIndex) {
 
 				return valueSwitch.getValueString(propertyInstance);
 
+			} else if (columnIndex == VERIFICATION_COLUMN) {
+
+				APropertyInstance propertyInstance = ca.getPropertyInstances().get(REQUIREMENT_VERIFICIATION_PROPERTY_NUMBER);
```
Comment (maurpa) [inline de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/provider/RequirementsAttributeLabelProvider.java:101]: Vertification_column and trace column is a duplicate just the return statement is different, here you could save some lines of code

**Review by JAmmermann-DLR — APPROVED** [review]
Approved

File: de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/provider/RequirementsAttributeLabelProvider.java (hunk 1 above)
Comment (franzTobiasDLR) [reply de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/provider/RequirementsAttributeLabelProvider.java:114]: Well, its always different properties with differnt XXX_PROPERTY_NUMBER. I moved the two lines in between to a new method now.

File: de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/provider/RequirementsAttributeLabelProvider.java (hunk 2 above)
Comment (franzTobiasDLR) [reply de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/provider/RequirementsAttributeLabelProvider.java:99]: Well, its always different properties with differnt XXX_PROPERTY_NUMBER. I moved the two lines in between to a new method now.

**Review by maurpa — APPROVED** [review]
Approved 👍

**Pull request merged**